apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/content/tab/repository/WaUiMetaDataRepository.java

```diff
@@ -1,23 +1,26 @@
 package gov.cdc.nbs.questionbank.page.content.tab.repository;
 
+import gov.cdc.nbs.questionbank.entity.WaTemplate;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Modifying;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 import gov.cdc.nbs.questionbank.entity.WaUiMetadata;
 
+import java.util.Optional;
+
 @Repository
 public interface WaUiMetaDataRepository extends JpaRepository<WaUiMetadata, Long> {
 
     @Query(value = "SELECT MAX(w.orderNbr) FROM WaUiMetadata w WHERE w.waTemplateUid.id =:page")
     Long findMaxOrderNumberByTemplateUid(@Param("page") Long page);
 
-    @Query(value = "select w.waTemplateUid from WaUiMetadata w where waUiMetadataUid = :waUiMetadataUid")
-    Long findPageNumber(@Param("waUiMetadataUid") Long waUiMetadataUid);
+    @Query(value = "select w.waTemplateUid from WaUiMetadata w where w.id = :id")
+    WaTemplate findPageNumber(@Param("id") Long id);
```

Comment:
Since this is returning the full `WaTemplate` now, i think you can remove this and use the default `findById`